app/models/rbac.rb

```diff
@@ -248,7 +248,33 @@ def self.find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_opti
     end
   end
 
+  def self.group(user_or_group)
+    case user_or_group
+    when User
+      user_or_group.current_group
+    when MiqGroup
+      user_or_group
+    when NilClass
```

Comment:
It may be better to add a method to User and MiqGroup and ultimately Tenant that returns the assigned tenant. This way the knowledge of how to get the tenant is within the corresponding class and you could get rid of this method all together.